ddprof-lib/src/main/cpp/profiler.cpp

```diff
@@ -183,7 +183,9 @@ inline u32 Profiler::getLockIndex(int tid) {
 }
 
 void Profiler::updateSymbols(bool kernel_symbols) {
+    Symbols::setupSignalHandlers();
```

Comment:
It would be nice to unify some of the signal handler code and add some tests around these tricky mechanisms.